components/Tag/Tag.jsx

```diff
@@ -38,37 +20,50 @@ class Tag extends Component {
 
   constructor() {
     super();
+
     this.state = {
-      isHover: false,
-      isRemoveHover: false,
-    };
-    this.setHover = this.setHover.bind(this);
-    this.setRemoveHover = this.setRemoveHover.bind(this);
+      isHovering:false,
+      isHoveringOverX:false
+    }
+
+    this.handleRemove = this.handleRemove.bind(this);
+    this.getXColor = this.getXColor.bind(this);
   }
 
-  setHover(value) {
-    return () => this.setState({ isHover: value });
+  handleRemove (event) {
+
+    console.log('click')
+    if(this.props.onRemove) {
```

Comment:
Tiny nitpick, but let's keep spaces after the `if`

so `if (this.props.onRemove)`